src/codegen/mod.rs

```diff
@@ -773,16 +769,25 @@ impl CodeGenerator for Type {
                         .expect("type parameters can always convert to rust ty OK")
                 }).collect();
 
-                if !params.is_empty() {
+                if params.is_empty() && ctx.options().rust_features.repr_transparent {
```

Comment:
I think this should be opt-in, or at least there should be a way to disable this without disabling `#[repr(transparent)]` altogether where it matters.